sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -62,12 +62,18 @@ export class BatchingReceiver extends MessageReceiver {
   /**
    * Receives a batch of messages from a ServiceBus Queue/Topic.
    * @param maxMessageCount      The maximum number of messages to receive.
+   * This has to be a number <= 2047 for receiver in `peekLock` mode, due to memory size
+   * constraints of underlying buffer.
    * @param idleTimeoutInSeconds The maximum wait time in seconds for which the Receiver
    * should wait to receive the first message. If no message is received by this time,
    * the returned promise gets resolved to an empty array.
    * @returns {Promise<ServiceBusMessage[]>} A promise that resolves with an array of Message objects.
    */
   receive(maxMessageCount: number, idleTimeoutInSeconds?: number): Promise<ServiceBusMessage[]> {
+    if (maxMessageCount > 2047) {
+      throw new Error("'maxMessageCount' cannot be > 2047 due to internal buffer limits.");
```

Comment:
Why throw an error?
We should still be able to return 2047 messages at max as per https://github.com/Azure/azure-sdk-for-js/issues/4748#issuecomment-521479607

Reply:
Also, this error message is irrelevant in case of ReceiveAndDelete mode which can handle more than 2047

Reply:
See other comment thread.. 
P.s If it's the same point/concern, then multiple/repeat comments can be avoided